src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -93,6 +94,22 @@ func (mod *mgmtModule) ID() drpc.ModuleID {
 // "DAOS_AGENT_DISABLE_CACHE=true" in the environment running the daos_agent.
 func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, error) {
 	var err error
+	pbReq := new(mgmtpb.GetAttachInfoReq)
+	if err := proto.Unmarshal(reqb, pbReq); err != nil {
+		return nil, drpc.UnmarshalingPayloadFailure()
+	}
+
+	mod.log.Debugf("GetAttachInfo req from client: %+v", pbReq)
+
+	if pbReq.Sys != mod.sys {
+		return nil, errors.Errorf("unknown system name %s", pbReq.Sys)
+	}
+
+	version := pbReq.GetVersion()
+	if version != mod.version {
+		return nil, errors.Errorf("get attach info client/agent protocol mismatch: client version: %#x, agent version %#x", pbReq.Version, mod.version)
```

Comment:
Since you have version as a variable, you could use that instead of pbReq.Version again.

Reply:
Fixed.